0000-route-broadcasting-protocol/RouteBroadcastingProtocol.asn

```diff
@@ -0,0 +1,68 @@
+RouteBroadcastingProtocol
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+  UInt8,
+  UInt16,
+  UInt32
+  FROM GenericTypes
+  Address
+  FROM InterledgerTypes
+;
+
+-- Route Control Request
+
+SyncMode ::= INTEGER {
+  idle (0),
+  sync (1)
+} (0..255)
+
+RoutingTableId ::= OCTET STRING (SIZE (16))
+
+Feature ::= OCTET STRING
+
+RouteControlRequestData ::= SEQUENCE {
+  syncMode SyncMode, -- the state that the node wants the counterpart node to be
+  routingTableId RoutingTableId, -- the routing table ID that the requesting node knows at that point
+  knownEpochIndex UInt32, -- the epoch that the requesting node knows at that point
+  features SEQUENCE OF Feature -- reserved for the future, currently not used
+}
+
+-- Route Update Request
+
+Auth ::= OCTET STRING (SIZE (32))
+
+-- meta (UInt8) is used as flags, and is reserved for the future.
+-- - 0x80 (1000 0000) : isOptional
+-- - 0x40 (0100 0000) : isTransitive
+-- - 0x20 (0010 0000) : isPartial
+-- - 0x10 (0001 0000) : isUtf8
+Property ::= SEQUENCE {
+  meta UInt8,
+  id UInt16,
+  value OCTET STRING
+}
+
+NewRoute ::= SEQUENCE {
+  addedRoutePrefix Address, -- the prefix of the added route
+  path SEQUENCE OF Address, -- the hops destinated to the prefix node
+  auth Auth, -- reserved for the future, currently not used
```

Comment:
It's not just reserved for the future! Routes are not yet filtered based on this but correct implementations will generate and correctly update the `auth` field.

Basically, the node originating a route will use a random value. Each node forwarding a route will SHA-256 hash this value.

It's not as good as BGPsec or other heavier schemes for routing security but it is better than nothing. An attacker can choose not to hash the auth value and pass it on unchanged, to make themselves look one hop closer than they are. But they cannot reverse the hash to make themselves look two or more hops closer. They can use an unrelated hash chain but connectors could be set such that this triggers a manual review.

In the future, we may add a routing PKI system to establish the correct base hash so that every knows which is the correct hash chain. Beyond that we may add full BGPsec style signing.

Reply:
@kruisdraad was discussing authenticated route updates in https://github.com/interledger/network/issues/4#issuecomment-427302643

My proposal to him was to look at using Distributed IDentifiers (DIDs) as a way to identify nodes and to root these in the distributed ledgers that we already use for settlement.

I have an XRP-biased proposal that could be adapted to be more generic: https://github.com/interledger/network/blob/master/docs/xrpl-pki.md

Reply:
I could not understand the whole security model of `auth`.
How does the receiver verify the auth value?

Reply:
I think what makes ILP beautiful is its independency. So I think it needs careful discussions to use some ledgers for the base of identities, that makes ILP dependent.